lib/spack/spack/schema/packages.py

```diff
@@ -82,6 +82,35 @@
                             },
                         },
                     },
+                    # If 'get_full_repo' is promoted to a Package-level
+                    # attribute, it could be useful to set it here
+                    "package_attributes": {
+                        "type": "object",
+                        "additionalProperties": False,
+                        "properties": {
+                            "git": {
+                                "type": "string",
+                            },
+                            "url": {
+                                "type": "string",
+                            },
+                            "submodules": {
+                                "type": "boolean",
+                            },
```

Comment:
Is there a reason these need to be special-cased? They would seem to fit in the schema below

Reply:
If you don't agree with https://github.com/spack/spack/pull/33275/files#r1012373337, then the do not need to be special-cased.